### PR TITLE
Decrypting a Solana Private Key and Signing a Tx within a Lit Action

### PR DESCRIPTION
**Pull request by spacesailor24**
See the provided `README` for instructions on how to run the example as well as a description of what's happening

### CODE REVIEW

**Comment by alexvotofuture** [comment]
- [ ]  User story/motivation for the guide has been identified
- [ ]  User story/motivation is addressed by the guide
- [ ]  Example code has contains all code and components required for a basic user to run it without implicit knowledge of other code/tools that need to be incorporated
- [ ]  Example code runs without bugs
- [ ]  No pseudocode
- [ ]  Full written explanations of all steps required to use a given feature (alongside code)
- [ ]  Warnings for when code is only relevant to a given testnet/function/version/etc.
- [ ]  Documentation titles and categorization are clear for new users
- [ ]  Guide and example code has been reviewed, followed, and successfully run by a peer who has not already worked on the guide/code and is taking on the persona of a target user for the user story/motivation
- [ ] Guide has been reviewed for correct spelling and grammar
- [ ]  Project Lead (or another knowledgeable party) for the feature has reviewed the guide/code and checked for accuracy/completeness
- [ ]  Final review and sign-off from Product Manager

**Comment by alexvotofuture** [comment]
- Is this intended to be an alternative to using Wrapped Keys, for cases when the user has the Solana key stored themselves somewhere without our servers and wants to do this all manually? Should we add a link to Wrapped Keys docs as an easier way to do this when that’s ready?
- Where is this going to be published? How will people discover this and know it’s what they need?
- Is there a guide associated with this example that explains the purpose and outlines the process?
- Probably should mention that the Solana account needs tokens in it first (for the tx and tx fees)
- Should we add comments to the code itself to explain what’s happening at each step? (I could infer from the console code what was happening, but might help to break things up with dividers for clarity)
- Should we break out the encryption of the Solana key to a separate code example? Or at least mention that the encryption would generally happen beforehand in this use case? (Otherwise, why would they need this example if they already had a Solana private key on their device and could already trigger the Solana tx right after the Ethereum tx manually?)
- Should we include optional Manzano/Habanero code so people can just run this with capacity credits without additional effort?

**Review by joshLong145 — APPROVED** [review]

**Comment by joshLong145** [comment]
Approving but we should address @alexvotofuture comments and make some readme updates.

File: lit-action-solana-decrypt-and-sign/nodejs/src/litAction.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  Connection,
+  Keypair,
+  PublicKey,
+  SystemProgram,
+  Transaction,
+} = require("@solana/web3.js");
+const bs58 = require("bs58");
+
+(async () => {
+  const solanaPrivateKey = await Lit.Actions.decryptAndCombine({
```
Comment (joshLong145) [inline lit-action-solana-decrypt-and-sign/nodejs/src/litAction.js:12]: can we add more error catching throughout this function and either use
- `setResponse`
- `console.log` (will get added to the response from `executeJs` in `logs`)

File: lit-action-solana-decrypt-and-sign/nodejs/src/litAction.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    accessControlConditions,
+    chain: "ethereum",
+    ciphertext,
+    dataToEncryptHash,
+    authSig: null,
+    sessionSigs,
```
Comment (DashKash54) [inline lit-action-solana-decrypt-and-sign/nodejs/src/litAction.js:18]: Why have we provided a sessionSig if authSig is null? It should take the authSig from the one provided while executing the Lit Action

**Review by DashKash54 — COMMENTED** [review]
Left a comment

File: lit-action-solana-decrypt-and-sign/nodejs/src/litAction.js (hunk 2 above)
Comment (DashKash54) [reply lit-action-solana-decrypt-and-sign/nodejs/src/litAction.js:18]: Also sessionSigs isn't even a param for this op_code: https://github.com/LIT-Protocol/lit-assets/pull/1167/files#diff-14e850f99e65a75544aa9f9ea5fd6c1f9833fdda0b8e43e930763f467c2170adR303

**Pull request closed without merging**